secureli/repositories/repo_files.py

```diff
@@ -82,13 +93,24 @@ def load_file(self, file_path: Path) -> str:
         if not file_path.exists() or not file_path.is_file():
             raise ValueError(f"File at path {file_path} did not exist")
 
-        if file_path.stat().st_size > self.max_file_size:
+        file_size = file_path.stat().st_size
+
+        if file_size > self.max_file_size:
             raise ValueError(f"File at path {file_path} was too big to scan")
 
         try:
-            with open(file_path, "r", encoding="utf8") as file_handle:
-                text = file_handle.read()
-                return text
+            with open(file_path, "rb") as file_handle:
+                data = file_handle.read()
+                encoding = chardet.detect(data)["encoding"]
+
+                # If resulting encoding is None, then it is binary
+                # Any file with zero size will be read as binary, so only skip binary files with size.
+                if encoding is None and file_size > 0:
+                    raise BinaryFileError(f"File at path {file_path} is a binary file")
+
+                return data.decode("utf-8")
+        except BinaryFileError as e:
+            raise e
```

Comment:
The logic for checking if it's binary makes sense to me, but does raising this error cause the LanguageAnalyzerService to break/fail? And if so is that the desired result or are expecting it to just skip these files without erroring?

Reply:
Got it, thanks for the clarification!